app/src/main/java/com/jbnm/homehero/ui/goal/GoalActivity.java

```diff
@@ -110,6 +168,62 @@ public void hideError() {
 
     @Override
     public void showRewardAnimation() {
+        Animation pulse = AnimationUtils.loadAnimation(this, R.anim.heart_pulse);
+        pulse.setAnimationListener(new Animation.AnimationListener() {
+            @Override
+            public void onAnimationStart(Animation animation) {}
+            @Override
+            public void onAnimationEnd(Animation animation) {
+                presenter.rewardAnimationEnded();
+            }
+            @Override
+            public void onAnimationRepeat(Animation animation) {}
+        });
+        rewardImageView.startAnimation(pulse);
+    }
+
+    @Override
+    public void showGoalPickerDialog() {
+        AlertDialog.Builder builder = new AlertDialog.Builder(context);
+        builder.setTitle(R.string.goal_picker_title)
+                .setAdapter(adapter, new DialogInterface.OnClickListener() {
+                    @Override
+                    public void onClick(DialogInterface dialogInterface, int i) {
+                        presenter.setNewRewardAndDecrementPoints(i);
+                    }
+                });
+        builder.create().show();
+    }
+
+    @Override
+    public void hideTaskButton() {
+        taskButton.setVisibility(View.GONE);
+    }
 
+    @Override
+    public void showTaskButton() {
+        taskButton.setVisibility(View.VISIBLE);
+    }
+
+    @Override
+    public void taskPickerIntent() {
+        Intent intent = new Intent(GoalActivity.this, TaskPickerActivity.class);
+        startActivity(intent);
+    }
+
+    @Override
+    public void taskProgressIntent() {
+        Intent intent = new Intent(GoalActivity.this, TaskProgressActivity.class);
```

Comment:
task progress needs the childId as an intent extra. TaskPicker has an example of this.